parity/configuration.rs

```diff
@@ -885,24 +885,23 @@ impl Configuration {
 	}
 
 	fn http_config(&self) -> Result<HttpConfiguration, String> {
-		let conf = HttpConfiguration {
-			enabled: self.rpc_enabled(),
-			interface: self.rpc_interface(),
-			port: self.args.arg_ports_shift + self.args.arg_rpcport.unwrap_or(self.args.arg_jsonrpc_port),
-			apis: self.rpc_apis().parse()?,
-			hosts: self.rpc_hosts(),
-			cors: self.rpc_cors(),
-			server_threads: match self.args.arg_jsonrpc_server_threads {
-				Some(threads) if threads > 0 => threads,
-				_ => 1,
-			},
-			processing_threads: self.args.arg_jsonrpc_threads,
-			max_payload: match self.args.arg_jsonrpc_max_payload {
-				Some(max) if max > 0 => max as usize,
-				_ => 5usize,
-			},
-			keep_alive: !self.args.flag_jsonrpc_no_keep_alive,
-		};
+		let mut conf = HttpConfiguration::default();
+		conf.enabled = self.rpc_enabled();
+		conf.interface = self.rpc_interface();
+		conf.port = self.args.arg_ports_shift + self.args.arg_rpcport.unwrap_or(self.args.arg_jsonrpc_port);
+		conf.apis = self.rpc_apis().parse()?;
+		conf.hosts = self.rpc_hosts();
+		conf.cors = self.rpc_cors();
+		if let Some(threads) = self.args.arg_jsonrpc_server_threads {
+			conf.server_threads = std::cmp::max(1, threads);
+		}
+		if let Some(threads) = self.args.arg_jsonrpc_threads {
+			conf.processing_threads = std::cmp::max(1, threads);
```

Comment:
It's actually fine to have `0` processing threads. This just means there is no additional cpu pool and the requests are processed directly on the server threads. I think that's possibly the best way to get the most performance - set the server threads to equal number of cores and processing threads to `0` so that there is no additional dispatch required.

However I feel that keeping `4` as default makes sense, cause it's especially important for transports that are single threaded by design (IPC or WS), without the processing threads we would have no concurrent processing at all.